jedi/api/environment.py

```diff
@@ -163,12 +163,26 @@ def find_python_environments():
                 pass
 
 
-def get_python_environment(python_name):
-    exe = find_executable(python_name)
-    if exe is None:
-        raise InvalidPythonEnvironment("This executable doesn't exist.")
-    path = os.path.dirname(os.path.dirname(exe))
-    return Environment(path, exe)
+# TODO: this function should probably return a list of environments since
+# multiple Python installations can be found on a system for the same version.
+def get_python_environment(python):
+    """
+    Return the first Python environment found for a given path or for a string
+    of the form 'pythonX.Y' where X and Y are the major and minor versions of
+    Python.
+    """
+    exe = find_executable(python)
+    if exe:
```

Comment:
Don't we need to differentiate here as well for Windows? AFAIK it's different for Windows than OS X/Linux. This probably only matters for the few cases where people actually provide a path. I think for Windows it's just one less `dirname` (I haven't really looked at `PC/getpathp.c`).